src/core/matchers/toHaveSize.js

```diff
@@ -0,0 +1,57 @@
+getJasmineRequireObj().toHaveSize = function(j$) {
+  /**
+   * {@link expect} the actual size to be equal to the expected, using array-like length or object keys size.
+   * @function
+   * @name matchers#toHaveSize
+   * @since 3.5.1
+   * @param {Object} expected - Expected size
+   * @example
+   * array = [1,2];
+   * expect(array).toHaveSize(2);
+   */
+  function toHaveSize(matchersUtil) {
+    return {
+      compare: function(actual, expected) {
+        var result = {
+            pass: false
+          },
+          diffBuilder = j$.DiffBuilder({prettyPrinter: matchersUtil.pp});
+
+        // Avoid misleading collections size matching
```

Comment:
I appreciate including the better error messages, but I'd like to see tests for this in the `toHaveSizeSpec`

Reply:
see ae340da2419ac960f16fd2ec1034f3dfe03e487a